envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsLoggingModeTest.kt

```diff
@@ -80,668 +85,683 @@ internal class IncomingPermissionsLoggingModeTest : EnvoyControlTestConfiguratio
-            path = "/log-unlisted-clients",
-            headers = mapOf("x-service-name" to "service-name-from-header"),
-            useSsl = true,
-            client = insecureClient
+        val echo2Response = echo2Envoy.ingressOperations.callLocalServiceInsecure(
+            endpoint = "/log-unlisted-clients",
+            headers = Headers.of(mapOf("x-service-name" to "service-name-from-header")),
+            useTls = true
         )
 
         // then
         assertThat(echo2Response).isOk().isFrom(echo2LocalService)
-        assertThat(echo2Envoy.ingressSslRequests).isOne()
-        assertThat(echo2Envoy).hasOneAccessDenialWithActionLog(
+        assertThat(echo2Envoy.container.ingressSslRequests).isOne()
+        assertThat(echo2Envoy.container).hasOneAccessDenialWithActionLog(
             protocol = "https",
             path = "/log-unlisted-clients",
             method = "GET",
             clientName = "service-name-from-header (not trusted)",
             trustedClient = false,
-            clientIp = echo2Envoy.gatewayIp()
+            clientIp = echo2Envoy.container.gatewayIp()
         )
     }
 
-    @BeforeEach
-    fun startRecordingRBACLogs() {
-        echoEnvoy.recordRBACLogs()
-        echo2Envoy.recordRBACLogs()
-    }
-
-    fun stopRecordingRBACLogs() {
-        echoEnvoy.logRecorder.stopRecording()
-        echo2Envoy.logRecorder.stopRecording()
-    }
+    @Test
+    fun `echo2 should NOT allow echo to access 'block-unlisted-clients-by-default' endpoint over https`() {
+        // when
+        val echo2Response =
+            echoEnvoy.egressOperations.callService("echo2", pathAndQuery = "/block-unlisted-clients-by-default")
 
-    @AfterEach
-    override fun cleanupTest() {
-        listOf(echoEnvoy, echo2Envoy, echo3Envoy).forEach { it.admin().resetCounters() }
-        stopRecordingRBACLogs()
+        // then
+        assertThat(echo2Response).isForbidden()
+        assertThat(echo2Envoy.container.ingressSslRequests).isOne()
+        assertThat(echo2Envoy.container).hasOneAccessDenialWithActionBlock(
+            protocol = "https",
+            path = "/block-unlisted-clients-by-default",
+            method = "GET",
+            clientName = "echo",
+            trustedClient = true,
+            clientIp = echoEnvoy.container.ipAddress()
+        )
     }
 
-    private fun callByProxy(proxy: String, path: String) = call(
-        address = proxy, pathAndQuery = path)
-
-    private fun callPost(service: String, from: EnvoyContainer, path: String) = call(
-        service = service, from = from, path = path, method = "POST",
-        body = RequestBody.create(MediaType.get("text/plain"), "{}"))
-
     private val EnvoyContainer.ingressSslRequests: Int?
-    get() = this.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()
+        get() = this.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()
 
     private val EnvoyContainer.ingressPlainHttpRequests: Int?
```

Comment:
Maybe lets replace this code with direct calls in test `envoy().container.admin().statValue("http.ingress_https.downstream_rq_completed")?.toInt()` (like in other tests)

Reply:
ok